### PR TITLE
Fix publishing of coverage results to coveralls.io (+ misc test config updates)

### PR DESCRIPTION
**Pull request by lukpueh**
**Fixes issue #**:
None

**Description of the changes being introduced by the pull request**:
Code coverage results from our Travis builds have not been published to coveralls.io in a while, due to the required `coveralls` publishing tool not being properly installed. 

Besides fixing publishing of coverage results, this PR cleans up/ updates other testing configurations:
 - Update travis to run on xenial
 - Add a build matrix to run each tox env in a corresponding travis env as per travis/tox best practices.
      https://docs.travis-ci.com/user/languages/python/#using-tox-as-the-build-script
 - Enable Python 3.5 tests, after all we claim to support Python 3.5
 - Remove "only build certain branch" restrictions
 - Use "install" instead of "before_script"  to install dependencies.
    Explicitly listing "install" prevents Travis from automatically running `pip install -r requirements.txt`, which is not necessary because most of those requirements are installed again in each  tox environment.
 - Move pylint and bandit calls to tox (pylint requires runtime dependencies to be installed).

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by lukpueh** [comment]
~**WIP! don't merge!**~

~Coveralls does run now, but it has problems reading the `.coverage` report...~

```
...
coverage.misc.CoverageException: Couldn't read data from '/home/travis/build/theupdateframework/tuf/tests/.coverage': CoverageException: Doesn't seem to be a coverage.py data file
...
coverage.misc.CoverageException: Couldn't read data from '/home/travis/build/theupdateframework/tuf/tests/.coverage': UnicodeDecodeError: 'utf-8' codec can't decode byte 0xa8 in position 31: invalid start byte
```

**Comment by lukpueh** [comment]
Fixed! See most recent [Travis builds](https://travis-ci.org/theupdateframework/tuf/builds/551203687) and corresponding [coverage reports on coveralls](https://coveralls.io/builds/24238801).

The badge on the main README should be updated once this PR is merged and IIUC we will see GitHub status (as in the table at the bottom) in subsequent PRs, because coveralls needs a reference build.

**Comment by lukpueh** [comment]
See most recent [Travis builds](https://travis-ci.org/secure-systems-lab/securesystemslib/builds/551256979?utm_source=github_status&utm_medium=notification) and corresponding [coverage reports on coveralls](https://coveralls.io/builds/24241109).

The badge on the main README should be updated once this PR is merged and IIUC we will see GitHub status (as in the table at the bottom) in subsequent PRs, because coveralls needs a reference build.

**Review by tanishqjasoria — APPROVED** [review]
I went through the PR and it looks good to me!

File: .travis.yml
Comment (adityasaky) [inline .travis.yml]: *is

**Review by adityasaky — CHANGES_REQUESTED** [review]

**Comment by lukpueh** [comment]
Thanks, @adityasaky!

**Pull request merged**
